modules/cef/format-cef-extension.c

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright (c) 2015 Balabit
+ * Copyright (c) 2015 bkil.hu
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 as published
+ * by the Free Software Foundation, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ */
+
+#include "filter/filter-expr.h"
+#include "filter/filter-expr-parser.h"
+#include "cfg.h"
+#include "value-pairs.h"
+#include "vptransform.h"
+#include "syslog-ng.h"
+#include "utf8utils.h"
+#include "format-cef-extension.h"
+
+typedef struct _TFCefState
+{
+  TFSimpleFuncState super;
+  ValuePairs *vp;
+} TFCefState;
+
+static gboolean
+tf_cef_prepare(LogTemplateFunction *self, gpointer s, LogTemplate *parent,
+		gint argc, gchar *argv[],
+		GError **error)
+{
+  TFCefState *state = (TFCefState *)s;
+
+  state->vp = value_pairs_new_from_cmdline (parent->cfg, argc, argv, error);
+  if (!state->vp)
+    return FALSE;
+
+  return TRUE;
+}
+
+typedef struct
+{
+  gboolean need_separator;
+  GString *buffer;
+  const LogTemplateOptions *template_options;
+} cef_state_t;
+
+static inline void
+g_string_append_escaped (GString *escaped_string, const char *str)
+{
+  const gchar *char_ptr = str;
```

Comment:
BTW we shouldn't use functions from the glib namespace, it was a bad idea even in the past, and this is a static function anyway so we are not forced to do that